src/screens/ConfigurePMTs/PaymentMethodConfig.res

```diff
@@ -0,0 +1,213 @@
+          )
+        } else {
+          enabledList.onChange(value->Identity.anyTypeToReactEvent)
+          enableType.onChange("enable_only"->Identity.anyTypeToReactEvent)
+        }
+      },
+      onFocus: _ev => (),
+      value: enabledList.value,
+      checked: true,
+    }
+    <SelectBox.BaseDropdown
+      allowMultiSelect=true
+      buttonText="Select Value"
+      input
+      options
+      hideMultiSelectButtons=true
+      showSelectionAsChips={false}
+      customButtonStyle="w-full"
+      fullLength=true
+      dropdownClassName={`${options->PaymentMethodConfigUtils.dropdownClassName}`}
+    />
+  }
+}
+
+let renderValueInp = (
+  options: array<SelectBox.dropdownOption>,
+  fieldsArray: array<ReactFinalForm.fieldRenderProps>,
+) => {
+  <PmtConfigInp options fieldsArray />
+}
+type valueInput = {
+  label: string,
+  name1: string,
+  name2: string,
+  options: array<SelectBox.dropdownOption>,
+}
+let valueInput = (inputArg: valueInput) => {
+  open FormRenderer
+  makeMultiInputFieldInfoOld(
+    ~label=`${inputArg.label}`,
+    ~comboCustomInput=renderValueInp(inputArg.options),
+    ~inputFields=[
+      makeInputFieldInfo(~name=`${inputArg.name1}`, ()),
+      makeInputFieldInfo(~name=`${inputArg.name2}`, ()),
+      makeInputFieldInfo(~name=`${inputArg.name2}.type`, ()),
+    ],
+    (),
+  )
+}
+
+@react.component
+let make = (
+  ~paymentMethodConfig: PaymentMethodConfigTypes.paymentMethodConfiguration,
+  ~config: string="",
+  ~setReferesh: unit => promise<unit>,
+  ~element: option<React.element>=?,
+) => {
+  open FormRenderer
+  let permissionJson = Recoil.useRecoilValueFromAtom(HyperswitchAtom.userPermissionAtom)
+  let (showPaymentMthdConfigModal, setShowPaymentMthdConfigModal) = React.useState(_ => false)
+  let (initialValues, setInitialValues) = React.useState(_ => Dict.make()->JSON.Encode.object)
+  let (currencies, setCurrencies) = React.useState(_ => [])
+  let (countries, setCountries) = React.useState(_ => [])
+  let (screenState, setScreenState) = React.useState(_ => PageLoaderWrapper.Success)
+  let {
+    payment_method_index,
+    payment_method_types_index,
+    connector_name,
+    payment_method_type,
+    merchant_connector_id,
+  } = paymentMethodConfig
+  open APIUtils
+  let connectorList = HyperswitchAtom.connectorListAtom->Recoil.useRecoilValueFromAtom
+  let fetchDetails = useGetMethod()
+  let updateDetails = useUpdateMethod()
+
+  let getProcessorDetails = async () => {
+    open LogicUtils
+    try {
+      setShowPaymentMthdConfigModal(_ => true)
+      setScreenState(_ => Loading)
+      let paymentMethoConfigUrl = getURL(~entityName=PAYMENT_METHOD_CONFIG, ~methodType=Get, ())
+      let data =
+        connectorList
+        ->Array.filter(item =>
+          item.merchant_connector_id === paymentMethodConfig.merchant_connector_id
+        )
+        ->Array.get(0)
+        ->Option.getOr(Dict.make()->ConnectorListMapper.getProcessorPayloadType)
```

Comment:
use getValueFromArray

Reply:
cannot use getValueFromArray because wanted to pass extra arguments to the mapper function